### DISCUSSION TITLE
is extracted Dovi metadata compatible with HDR10+?

### DISCUSSION THREAD

**Discussion by wyup** [quietvoid/hdr10plus_tool#67] [General]
Hello,
I would like to know if I can inject a Dovi extracted .json metadata into x265 to make it HDR10 compatible:

`hdr10plus_tool inject -i video.hevc -j metadata.json -o injected_output.hevc`

or using ffmpeg:

`ffmpeg -fflags +discardcorrupt -ec guess_mvs+deblock+favor_inter -i input.file -c:a ac3 -ac 2 -ar 48000 -c:v libx265 -preset medium -x265-params 'hdr-opt=1:repeat-headers=1:colorprim=bt2020:transfer=smpte2084:colormatrix=bt2020nc:master-display=G(x,y)B(x,y)R(x,y)WP(x,y)L(x,y):max-cll=x,y:hdr10=1:dhdr10-info=output.json**' -map_metadata -1 -map_chapters -1 output.file`

**Comment by quietvoid** [comment]
No, you can't do that. They are two completely different JSON.

**Reply by wyup** [reply]
Thanks. Is there anything Samsung (HDR10+ compatible, non Dovi) tv users can rescue from Dovi file-based content to improve playback?

Why is it taking so long to reverse-engineer Dovi to transcode to open-source HDR10+?

Does Dovi embed color grading to the scene apart from pure dynamic luminance ranges?

**Reply by quietvoid** [reply]
> Why is it taking so long to reverse-engineer Dovi to transcode to open-source HDR10+?

It's not complicated, just not a straight forward conversion: https://github.com/quietvoid/dovi_tool/issues/260#issuecomment-1780971652

**Reply by wyup** [reply]
> HDR10+ needs the brightness histogram, which isn't possible to compute solely from Dolby Vision metadata.
> It would always require measuring the actual video brightness, which is out of scope for this project.

It looks like it would not be complicated to make a script that measures a histogram from a video file. Does ffmpeg or any open source tool decode DoVi? I think Android has a license.

**Reply by quietvoid** [reply]
FFmpeg supports it through libplacebo.

**Reply by wyup** [reply]
When you say 'measuring the actual video brightness', do you mean parsing the whole video, logging per-scene or per-frame brightness values, process that and then create a new HDR10+ compatible metadata from scratch, discarding previous DV metadata?

**Reply by quietvoid** [reply]
> do you mean parsing the whole video, logging per-scene or per-frame brightness values, process that and then create a new HDR10+ compatible metadata from scratch, discarding previous DV metadata?

Yes that's how it would be done.
The only metadata that could be used from Dolby Vision are the shot boundaries.

**Reply by wyup** [reply]
I see, but that is too much work i guess, it looks more practical to use a HDR10+ content production suite, such as Davinci Resolve: it auto detects shot boundaries. But then there is DV license for decoding.

It is a pity Samsung is alone with HDR10+. Platforms are discarding it for DV. Even when HDR10+ is an open standard, and equally capable. It should be mandatory for BD HDR-UHD, so all TV manufacturers included it for free. Only Google encourages patent-free technology with AV1/VP9 and Opus. DV is a blackbox and they introduce all kinds of image processing there, locking their technology.

**Reply by quietvoid** [reply]
HDR10+ isn't truly open, the specifications are not available publicly.
And it isn't equally capable either.

Anyways I digress. It would be feasible and I do plan on having HDR10+ generation in my VapourSynth code.
Dolby Vision can also be generated without proprietary grading software.

**Reply by wyup** [reply]
With scene detection and measured brightness data (by either Resolve or VS script), how would HDR10+ generated dynamic metadata differ in result from DV's own? Is it automatically calculated or manually graded by a colorist?
I think there is a L2 pass in DV that is done by a colorist i.e. to lift up detail in dark scenes, and I don't know if this is included in the dynamic metadata or it is an additional DV layer.

**Reply by quietvoid** [reply]
It's automatically calculated but the colorist can also make adjustments, so it's not equivalent to HDR10+ which is always automatically generated.

**Comment by wyup** [comment]
> FFmpeg supports it through libplacebo.

Does libplacebo decode DoVi dynamic metadata aswell?
Videolan [Libplacebo](https://code.videolan.org/videolan/libplacebo) Readme states "_Native support for Dolby Vision HDR, including Profile 5 conversion to HDR/PQ or SDR, reading DV **side data**, and reshaping. (BL only, currently)_"

**Reply by quietvoid** [reply]
No, it only does brightness metadata (L1) to tone map.

**Comment by wyup** [comment]
> It would be feasible and I do plan on having HDR10+ generation in my VapourSynth code.

How's that doing? It would be great to port L1 brightness metadata to HDR10+, or instead generate it as you said. This would be ideal to port DV movies to HDR10+ for Samsung tvs and enhance low light scenes. I know i can generate HDR10+ from Resolve but I'd prefer an elegant script tool :-)

**Comment by quietvoid** [comment]
I've implemented the HDR10+ generation in `awsmfunc` some time ago: https://github.com/OpusGang/awsmfunc/blob/master/src/awsmfunc/hdr10plus.py#L12
It is for VapourSynth and is basic HDR10+ with histogram metadata only. It has no Bezier curve.

**Reply by wyup** [reply]
Cool. Is it used by any script or application?

**Reply by quietvoid** [reply]
Not that I'm aware of.
